app/src/main/java/com/onewelcome/showcaseapp/feature/userderegistration/UserDeregistrationViewModel.kt

```diff
@@ -0,0 +1,78 @@
+package com.onewelcome.showcaseapp.feature.userderegistration
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.onFailure
+import com.github.michaelbull.result.onSuccess
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onewelcome.core.usecase.DeregisterUserUseCase
+import com.onewelcome.core.usecase.GetUserProfilesUseCase
+import com.onewelcome.core.usecase.IsSdkInitializedUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class UserDeregistrationViewModel @Inject constructor(
+  private val isSdkInitializedUseCase: IsSdkInitializedUseCase,
+  private val getUserProfilesUseCase: GetUserProfilesUseCase,
+  private val deregisterUserUseCase: DeregisterUserUseCase
+) : ViewModel() {
+
+  var uiState by mutableStateOf(State())
+    private set
+
+  fun onEvent(event: UiEvent) {
+    when (event) {
+      is UiEvent.LoadInitialData -> loadInitialData()
+      is UiEvent.OnUserProfileSelected -> uiState = uiState.copy(selectedUserProfile = event.userProfile)
+      is UiEvent.DeregisterUser -> deregisterUser()
+    }
+  }
+
+  private fun loadInitialData() {
+    val isSdkInitialized = isSdkInitializedUseCase.execute()
+    uiState = uiState.copy(isSdkInitialized = isSdkInitialized)
+    if (isSdkInitialized) {
+      loadUserProfiles()
+    }
+  }
+
+  private fun loadUserProfiles() {
+    viewModelScope.launch {
+      getUserProfilesUseCase.execute()
+        .onSuccess { uiState = uiState.copy(registeredUserProfiles = it, selectedUserProfile = it.firstOrNull()) }
+        .onFailure { uiState = uiState.copy(registeredUserProfiles = emptySet(), selectedUserProfile = null) }
+    }
+  }
+
+  private fun deregisterUser() {
+    viewModelScope.launch {
+      val result = uiState.selectedUserProfile?.let {
+        deregisterUserUseCase.execute(it)
+      } ?: Err(IllegalArgumentException("User profile not selected"))
+      uiState = uiState.copy(isLoading = true)
```

Comment:
uiState = uiState.copy(isLoading = true) 

this line should be higher, first one in the method